core/unit_test/TestReduceCombinatorical.hpp

```diff
@@ -414,10 +414,15 @@ struct TestReduceCombinatoricalInstantiation {
 #endif
   }
 
+  struct AddIndexToSum {
+    KOKKOS_FUNCTION
+    void operator()(const int i, double& lsum) const { lsum += i; }
+  };
+
   template <class... Args>
   static void AddLambdaRange(int N, void*, Args... args) {
     AddReturnArgument(
-        N, args..., KOKKOS_LAMBDA(const int& i, double& lsum) { lsum += i; });
+        N, args..., AddIndexToSum{});
```

Comment:
@crtrott @dalg24 This is a very weird bug, see my `cuda-gdb` stack trace.

My first though was to simplify the trace output by relying on some `struct` instead of a lambda. And it seemed to fix the segfault.

I don't know why, and I guess it would be important to put some man power into this (not mine though :laughing:)